### PR TITLE
Overhauled deployment events and logs

### PR DESCRIPTION
**Pull request by rauchg**
This PR
- Increases the reliability of the deployment output in the event of networking issues
- Implements our upcoming deployment events API
- Suppresses HTTP/2 warnings
- Makes some cosmetic changes and improvements
- Improves instrumentation

Hasn't been tested thoroughly yet, but it's ready for early reviews

### CODE REVIEW

File: src/providers/sh/commands/deploy.js [hunk 1]
```diff
@@ -782,6 +791,89 @@ async function readMeta(
...
+          // and ignore everything else
+          if (quiet) {
+            if (type === 'instance-start') {
+              resolve();
+            }
+            return;
+          }
+
+          switch (type) {
+            case 'build-start':
```
Comment (rauchg) [inline src/providers/sh/commands/deploy.js]: This is waiting on an API improvement by @arunoda

Comment (arunoda) [reply src/providers/sh/commands/deploy.js]: @rauchg here's the fix: https://github.com/zeit/now-atlas-events/pull/16 
Also krolled this into BRU.

**Comment by Qix-** [comment]
lgtm

**Review by Qix- — APPROVED** [review]

**Comment by leo** [comment]
Sweet! Going out on canary in a sec.

**Pull request merged**